### PR TITLE
GH-166: Add BindingsEndpoint to FAQ

### PR DESCRIPTION
**Pull request by garyrussell**

### CODE REVIEW

File: content/documentation/pages/7-resources/3-faq.md [hunk 1]
```diff
@@ -417,6 +417,40 @@ See the [compatibility matrix](https://github.com/spring-cloud/spring-cloud-stre
...
 
+<!--QUESTION#lifecycle-->
+
+Can I manage binding lifecycles?
+
+By default, bindings are started automatically when the application is initialized.
+Bindings implement the Spring `SmartLifecycle` interface.
+`SmartLifecycle` allows beans to be started in phases.
+Producer bindings are started in an early phase (`Integer.MIN_VALUE + 1000`).
+Consumer bindings are started in a late phase (`Integer.MAX_VALUE - 1000`).
```
Comment (olegz) [inline content/documentation/pages/7-resources/3-faq.md]: I wonder if we need to mention `Integer.MIN_VALUE + 1000`. Could be distracting `early/late` is more then enough IMHO.

File: content/documentation/pages/7-resources/3-faq.md [hunk 2]
```diff
@@ -417,6 +417,40 @@ See the [compatibility matrix](https://github.com/spring-cloud/spring-cloud-stre
...
+This leaves room in the spectrum such that user beans implementing `SmartLifecycle` can be started before producer bindings, after consumer bindings, or anywhere in between.
+
```
Comment (olegz) [inline content/documentation/pages/7-resources/3-faq.md]: If agree with above, then this whole sentence is no longer relevant.

File: content/documentation/pages/7-resources/3-faq.md [hunk 3]
```diff
@@ -417,6 +417,40 @@ See the [compatibility matrix](https://github.com/spring-cloud/spring-cloud-stre
...
+You can disable auto-startup by setting the consumer or producer `autoStartup` property to `false`.
+
+Binding lifecycles can be visualized and controlled using Boot actuators; see [Binding visualization and control](https://cloud.spring.io/spring-cloud-static/spring-cloud-stream/current/reference/html/spring-cloud-stream.html#binding_visualization_control).
+
```
Comment (olegz) [inline content/documentation/pages/7-resources/3-faq.md]: Let's link to https://cloud.spring.io/spring-cloud-static/spring-cloud-stream/3.0.0.RELEASE/reference/html/spring-cloud-stream.html#binding_visualization_control as we are working on making `current` point to that.

File: content/documentation/pages/7-resources/3-faq.md (hunk 1 above)
Comment (garyrussell) [reply content/documentation/pages/7-resources/3-faq.md]: Well, we need to tell them how they can ensure their bean(s) can be started before producers or after consumers; so if we just say "early" and "late" we'd then need to say something like.

>To start a bean before the producer binding, it must be in phase less than `Integer.MIN_VALUE - 1000` 

(and a similar sentence for starting beans after consumer bindings.

So, I am not sure what the benefit would be.

File: content/documentation/pages/7-resources/3-faq.md (hunk 2 above)
Comment (garyrussell) [reply content/documentation/pages/7-resources/3-faq.md]: I think this is more succinct.

File: content/documentation/pages/7-resources/3-faq.md (hunk 3 above)
Comment (olegz) [reply content/documentation/pages/7-resources/3-faq.md]: Strike that as it has been done already.

Comment (garyrussell) [reply content/documentation/pages/7-resources/3-faq.md]: I specifically used `current` so the link stays up to date - we don't want to point to old docs when, say, we go to 3.1 etc.

File: content/documentation/pages/7-resources/3-faq.md (hunk 1 above)
Comment (olegz) [reply content/documentation/pages/7-resources/3-faq.md]: I agree with that, but that would be rather very advance concept and in the spirit of _'less is more'_  I'd rather know why a user wants to start/stop something deliberately before/after etc. . .

I am good either way, just throwing it out there to see if any one else has an opinion about that.

File: content/documentation/pages/7-resources/3-faq.md (hunk 3 above)
Comment (olegz) [reply content/documentation/pages/7-resources/3-faq.md]: Right, but the 'current' didn't work properly until recently.

Comment (garyrussell) [reply content/documentation/pages/7-resources/3-faq.md]: I checked it before I changed it from 3.0.0 😄

**Pull request merged**